test/archethic/contracts/interpreter/scope_test.exs

```diff
@@ -5,55 +5,589 @@ defmodule Archethic.Contracts.Interpreter.ScopeTest do
 
   doctest Scope
 
-  test "global variables" do
-    Scope.init(%{"var1" => %{"prop" => 1}})
-    Scope.write_at([], "var2", 2)
-    assert Scope.read_global(["var1", "prop"]) == 1
-    assert Scope.read_global(["var2"]) == 2
+  describe "init/1" do
+    test "should instantiate Scope with empty context list and global vars" do
+      constants = %{"var1" => %{"prop" => 1}, "var2" => 4}
+      Scope.init(constants)
+      assert %{"context_list" => [], "var1" => %{"prop" => 1}, "var2" => 4} == Process.get(:scope)
+    end
   end
 
-  test "write_at/3" do
-    Scope.init()
+  describe "create_context/0" do
+    test "create_context/0 should create empty context to existing scope" do
+      Process.put(:scope, %{"context_list" => [], "var1" => %{"prop" => 1}, "var2" => 4})
 
-    hierarchy = ["depth1"]
-    Scope.create(hierarchy)
-    Scope.write_at(hierarchy, "var1", 1)
-    assert Scope.read(hierarchy, "var1") == 1
+      Scope.create_context()
 
-    hierarchy = ["depth1", "depth2"]
-    Scope.create(hierarchy)
-    Scope.write_at(hierarchy, "var2", 2)
-    assert Scope.read(hierarchy, "var2") == 2
+      %{"context_list" => [context_ref]} = Process.get(:scope)
+
+      assert %{
+               "context_list" => [^context_ref],
+               "var1" => %{"prop" => 1},
+               "var2" => 4,
+               ^context_ref => %{
+                 "scope_hierarchy" => []
+               }
+             } = Process.get(:scope)
+    end
+
+    test "create_context/0 should create new context if one already exists" do
+      # create scope with existing context
+      Process.put(:scope, %{
+        "context_list" => ["context_1"],
+        "var1" => %{"prop" => 1},
+        "var2" => 4,
+        "context_1" => %{
+          "scope_hierarchy" => []
+        }
+      })
+
+      Scope.create_context()
+
+      %{"context_list" => [context_ref_2, _]} = Process.get(:scope)
```

Comment:
Same here for assert.
You can also match that second arg of the list is the first context 
```elixir
assert %{"context_list" => [context_ref_2, "context_1"]} = Process.get(:scope)
```